components/component.go

```diff
@@ -1,20 +1,21 @@
 package components
 
 import (
+	operatorv1 "github.com/openshift/api/operator/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 type Component struct {
-	// Set to "true" to enable component, "false" to disable component. A disabled component will not be installed.
-	Enabled bool `json:"enabled"`
+	// Set to "managed" to enable the component, and to "removed" to disable it.
```

Comment:
```suggestion
	// Set to "managed" to enable the component, and to "removed" to disable it.
	// +kubebuilder:validation:Enum=managed;removed
```
to set enum in list, if user input anything else wont be accepted